### PR TITLE
Configure and handle redirection on login and logout

### LINKED ISSUE #123: Handle redirection after successful login
**Is your feature request related to a problem? Please describe.**
Currently after using the LinkedIn sign in user is not redirected back to the frontend application, but it is required to be redirected back to the exact URL the user was, where the login button was clicked 

**Describe the solution you'd like**
Change spring-security config and define a `successHandler` to handle redirection

### LINKED ISSUE #124: Configure and Handle redirection after successful logout
**Is your feature request related to a problem? Please describe.**
Currently, the logout feature isn't properly configured and it is required the users to be redirected back to the homepage, where the logout button is clicked 

**Describe the solution you'd like**
Change spring-security config and define a `logoutSuccessHandler` to handle redirection

### PR DESCRIPTION
**Pull request by Gravewalker666**
## Purpose

The purpose of this PR is to fix #123 fix #124 

## Goals
- To make users redirect to where they were before using the login feature
- To configure a logout feature

## Approach
- Made a new endpoint for `login` and saved the `referer` Header value to a session variable
- Redirected the user to the auth process 
- Created a function to handle the redirection after successful login by taking the previously saved session variable value and redirecting the user back to the same page 
- Used the `referer` and redirected users back to the home page after successful logout

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
- Apache Maven 3.6.3
- JDK Oracle 8
- macOS, BigSur

## Learning
- https://stackoverflow.com/questions/14573654/spring-security-redirect-to-previous-page-after-successful-login

### CODE REVIEW

File: src/main/java/org/sefglobal/scholarx/config/SecurityConfig.java [hunk 1]
```diff
@@ -47,6 +52,15 @@ protected void configure(HttpSecurity http) throws Exception {
       .permitAll()
       .and()
       .logout()
+            .logoutSuccessHandler(new LogoutSuccessHandler() {
+              @Override
+              public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse httpServletResponse, Authentication authentication) throws IOException, ServletException {
```
Comment (YohanAvishke) [inline src/main/java/org/sefglobal/scholarx/config/SecurityConfig.java:57]: break this line to not increase the character limit of a line

**Review by YohanAvishke — CHANGES_REQUESTED** [review]
Good job. Only a minor fix is required

File: src/main/java/org/sefglobal/scholarx/config/SecurityConfig.java (hunk 1 above)
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/config/SecurityConfig.java:57]: Sure, on it

Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/config/SecurityConfig.java]: Resolved with the commit a074c37 Please Review

**Pull request merged**